app/controllers/student_requests_controller.rb

```diff
@@ -62,6 +65,9 @@ def edit
   # end
   
   def adminview
+    if session_get(:uin) == nil
+      redirect_to root_path
+    end
```

Comment:
Better to have if else when using redirect_to. The control also continues beyond redirect_to and may cause inconsistency.